src/table.rs

```diff
@@ -480,10 +554,129 @@ mod tests {
+        assert_eq!(table_sum, table.players.len() as i32 * DEFAULT_START_MONEY);
+    }
+
+    #[test]
+    pub fn test_side_pot() {
+        let mut table = deal_test_cards_tied_best_side_pot();
+        table.current_player_index = 0;
+        table.dealer_button_index = table.players.len() - 1;
+        table.update_current_player_index_to_next_active();
+        assert_eq!(2, table.current_player_index);
+        table.take_action(HandAction::Raise(1));
+        assert_eq!(3, table.current_player_index);
+        table.take_action(HandAction::Call);
+        // assert_eq!(1, table.current_player_index);
+        // table.take_action(HandAction::Call);
+        // assert_eq!(2, table.current_player_index);
+        // table.take_action(HandAction::Call);
+        for i in 2..4 {
+            assert_eq!(table.table_state, Flop);
+            assert_eq!(table.current_player_index, i);
+            table.take_action(HandAction::Check);
+        }
+        for i in 2..4 {
+            assert_eq!(table.table_state, Turn);
+            assert_eq!(table.current_player_index, i);
+            table.take_action(HandAction::Check);
+        }
+        for i in 2..4 {
+            assert_eq!(table.table_state, River);
+            assert_eq!(table.current_player_index, i);
+            table.take_action(HandAction::Check);
+        }
+        assert_eq!(table.table_state, PreFlop);
+        assert!(table.players[0].is_alive());
+        assert!(table.players[1].is_alive());
+        assert_eq!(table.players[0].total_money, 2);
+        assert_eq!(table.players[1].total_money, 2);
+        assert_eq!(table.players[2].total_money, 499);
+        assert_eq!(table.players[3].total_money, 497);
+        assert_eq!(table.players[4].total_money, 498);
+        assert_eq!(table.players[5].total_money, 498);
+    }
+
+    #[test]
+    pub fn one_winner() {
+        // Required for the table evaluator
+        let mut table = deal_test_cards();
+        check_table_has_right_amount(&table);
+        table.resolve_hand();
+        for player in &table.players {
+            assert!(player.is_alive());
+        }
+        check_table_has_right_amount(&table);
+        assert_eq!((table.players.len() as i32 - 2) * table.ante + DEFAULT_START_MONEY, table.players.get(0).unwrap().total_money);
+    }
+
+    #[test]
+    pub fn two_winners() {
+        // Required for the table evaluator
+        let mut table = deal_test_cards_tied_best();
+        check_table_has_right_amount(&table);
+        table.resolve_hand();
+        for player in &table.players {
+            assert!(player.is_alive());
+        }
+        check_table_has_right_amount(&table);
+        assert_eq!((table.players.len() as i32 - 1) * table.ante / 2 - table.ante + DEFAULT_START_MONEY, table.players.get(0).unwrap().total_money);
+        assert_eq!((table.players.len() as i32 - 1) * table.ante / 2 - table.ante + DEFAULT_START_MONEY, table.players.get(1).unwrap().total_money);
```

Comment:
I think there's some rounding down happening here, but it would give us the right number. Would it make more sense for the left side of this to be something like 
(table.players.len() as i32) * table.ante / 2 - (table.ante * 2) + DEFAULT_START_MONEY
?
If I'm reading this correctly, that makes it more clear that the subtraction is because the players have ante'd twice. Could probably do something similar for the test above as well, it's not clear why we're subtracting from the number of players instead of doing `* table.ante` first. I know table.ante is 1 so it doesn't really matter, but I think that would be more clear.

Reply:
Yeah I think I was just doing algebra but this is a lot easier to read I think. It's updated to yours now